app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,99 @@
+class TasksController < ApplicationController
+
+  # THE DECENDING ORDER SORT ON DATE_DUE IS NOT WORKING IN RAILS CONSOLE; DOES WORK IN THE BROWSER
+
+  def index
+    @tasks = Task.all.order(date_due: :asc)
+  end
+
+  def show
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+
+    unless @task
+      redirect_to tasks_path
+    end
+  end
+
+  def new
+    @task = Task.new(task_name: "to-do")
+  end
+
+  def create
+    this_task = Task.new(task_params)
+    is_successful = this_task.save
+
+    if is_successful
+      redirect_to tasks_path
+    else
+      head :not_found
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    unless @task
+      redirect_to tasks_path
+    end
+  end
+
+  def update
+    this_task = Task.find_by(id: params[:id])
+
+    is_successful = this_task.update(task_params)
+
+    if is_successful
+      redirect_to task_path
+    else
+      redirect_to tasks_path
+    end
+  end
+
+  def destroy
+    this_task = Task.find_by(id: params[:id])
+
+    is_successful = this_task.destroy
+
+    if is_successful
+      redirect_to tasks_path
+    else
+      head :not_found
+    end
+  end
+
+  def toggle_completed
+    this_task = Task.find_by(id: params[:id])
+    # raise
+    # this_task.toggle(:completed)
+    # this_task[:date_completed] = Time.now
+    # this_task.save
+
+    if this_task.completed == true
+      this_task.completed = false
+      this_task[:date_completed] = nil
+    elsif this_task.completed == false
+      this_task.completed = true
+      this_task[:date_completed] = Time.now
+    end
```

Comment:
Great work on this logic! It's accurate and has the right functionality. It can probably be refactored to something a little more slim. Don't forget-- `if this_task.completed == true` is the same thing as `if this_task.completed`, and you could use an `else` instead of `elsif ...`!